packages/contracts/contracts/minter-suite/Minters/MinterDALinV5.sol

```diff
@@ -0,0 +1,548 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../../interfaces/v0.8.x/IGenArt721CoreContractV3_Base.sol";
+import "../../interfaces/v0.8.x/IDelegationRegistry.sol";
+import "../../interfaces/v0.8.x/ISharedMinterV0.sol";
+import "../../interfaces/v0.8.x/ISharedMinterDAV0.sol";
+import "../../interfaces/v0.8.x/ISharedMinterDALinV0.sol";
+import "../../interfaces/v0.8.x/IMinterFilterV1.sol";
+
+import "../../libs/v0.8.x/minter-libs/SplitFundsLib.sol";
+import "../../libs/v0.8.x/minter-libs/MaxInvocationsLib.sol";
+import "../../libs/v0.8.x/minter-libs/DALib.sol";
+import "../../libs/v0.8.x/AuthLib.sol";
+
+import "@openzeppelin-4.5/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin-4.5/contracts/utils/math/SafeCast.sol";
+
+pragma solidity 0.8.19;
+
+/**
+ * @title Filtered Minter contract that allows tokens to be minted with ETH.
+ * This is designed to be used with GenArt721CoreContractV3 flagship or
+ * engine contracts.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ */
+contract MinterDALinV5 is
+    ReentrancyGuard,
+    ISharedMinterV0,
+    ISharedMinterDAV0,
+    ISharedMinterDALinV0
+{
+    using SafeCast for uint256;
+    /// Minter filter address this minter interacts with
+    address public immutable minterFilterAddress;
+
+    /// Minter filter this minter may interact with.
+    IMinterFilterV1 private immutable minterFilter;
+
+    /// minterType for this minter
+    string public constant minterType = "MinterDALinV5";
+
+    /// minter version for this minter
+    string public constant minterVersion = "v5.0.0";
+
+    uint256 constant ONE_MILLION = 1_000_000;
+    /// Minimum auction length in seconds
+    uint256 public minimumAuctionLengthSeconds = 3600;
+
+    /// contractAddress => projectId => base project config
+    mapping(address => mapping(uint256 => ProjectConfig))
+        private _projectConfigMapping;
```

Comment:
this can be moved entirely for this minter (as well as `function projectConfig`)